lib/helpers/cookieHelper.js

```diff
@@ -1,68 +1,44 @@
-/*
- * catberry-oauth
- *
- * Copyright (c) 2014 Denis Rechkunov and project contributors.
- *
- * catberry-oauth's license follows:
- *
- * Permission is hereby granted, free of charge, to any person
- * obtaining a copy of this software and associated documentation
- * files (the "Software"), to deal in the Software without restriction,
- * including without limitation the rights to use, copy, modify, merge,
- * publish, distribute, sublicense, and/or sell copies of the Software,
- * and to permit persons to whom the Software is furnished to do so,
- * subject to the following conditions:
- *
- * The above copyright notice and this permission notice shall be included
- * in all copies or substantial portions of the Software.
- *
- * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
- * OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
- * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
- * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
- * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
- * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
- * THE SOFTWARE.
- *
- * This license applies to all parts of catberry-oauth that are not externally
- * maintained libraries.
- */
-
 'use strict';
 
+const COOKIE_CLEAN_REGEXP = /^"|"$/g;
+const COOKIE_SPLIT_REGEXP = /; */;
+
 module.exports = {
+
 	/**
 	 * Gets cookie from HTTP request headers.
 	 * @param {IncomingMessage} request HTTP request.
 	 * @returns {Object} Map of all cookie values.
 	 */
-	getFromRequest: function (request) {
+	getFromRequest(request) {
 		if (!request.headers || typeof (request.headers.cookie) !== 'string') {
 			return {};
 		}
 
-		var result = {};
+		const result = {};
 		request.headers.cookie
-			.split(/; */)
-			.forEach(function (cookiePair) {
-				var equalsIndex = cookiePair.indexOf('=');
+			.split(COOKIE_SPLIT_REGEXP)
+			.forEach(cookiePair => {
+				const equalsIndex = cookiePair.indexOf('=');
 				if (equalsIndex < 0) {
 					return;
 				}
 
-				var key = cookiePair.substr(0, equalsIndex).trim(),
-					value = cookiePair.substr(
-						equalsIndex + 1, cookiePair.length
-					).trim();
+				const key = cookiePair.substr(0, equalsIndex).trim();
+				let value = cookiePair.substr(equalsIndex + 1, cookiePair.length).trim();
 
-				value = value.replace(/^"|"$/g, '');
+				value = value.replace(COOKIE_CLEAN_REGEXP, '');
```

Comment:
No, without the `g` option it won't work, just don't use a global RegExp in this case it should be an in-place literal as it was before.